test/extensions/Worker.Extensions.Http.AspNetCore.Tests/RegistrationExpectedInAspNetIntegrationTests.cs

```diff
@@ -0,0 +1,315 @@
+    {
+        private const string ExpectedRegistrationMethod = "ConfigureFunctionsWebApplication";
+
+        [Fact]
+        public async Task AspNetIntegration_MissingRegistration_Diagnostics_Expected()
+        {
+            string testCode = @"
+                namespace AspNetIntegration
+                {
+                    using System.Linq;
+                    using System.Threading.Tasks;
+                    using Microsoft.Azure.Functions.Worker;
+                    using Microsoft.Extensions.DependencyInjection;
+                    using Microsoft.Extensions.Hosting;
+                    using Microsoft.Extensions.Logging;
+
+                    class Program
+                    {
+                        static void Main(string[] args)
+                        {
+                            var host = new HostBuilder()
+                                .ConfigureFunctionsWorkerDefaults()
+                                .Build();
+
+                            host.Run();
+                        }
+                    }
+                }";
+
+            var test = new AnalyzerTest
+            {
+                ReferenceAssemblies = LoadRequiredDependencyAssemblies(),
+                TestCode = testCode
+            };
+
+            test.ExpectedDiagnostics.Add(Verifier.Diagnostic()
+                            .WithSeverity(Microsoft.CodeAnalysis.DiagnosticSeverity.Warning)
+                            .WithSpan(13, 37, 13, 41)
+                            .WithArguments(ExpectedRegistrationMethod));
+
+            await test.RunAsync();
+        }
+
+
+        [Fact]
+        public async Task AspNetIntegration_CommentedRegistration_Diagnostics_Expected()
+        {
+            string testCode = @"
+                namespace AspNetIntegration
+                {
+                    using System.Linq;
+                    using System.Threading.Tasks;
+                    using Microsoft.Azure.Functions.Worker;
+                    using Microsoft.Extensions.DependencyInjection;
+                    using Microsoft.Extensions.Hosting;
+                    using Microsoft.Extensions.Logging;
+
+                    class Program
+                    {
+                        static void Main(string[] args)
+                        {
+                            var host = new HostBuilder()
+                                //.ConfigureFunctionsWorkerDefaults()
+                                //.ConfigureFunctionsWebApplication()
+                                .Build();
+
+                            host.Run();
+                        }
+                    }
+                }";
+
+            var test = new AnalyzerTest
+            {
+                ReferenceAssemblies = LoadRequiredDependencyAssemblies(),
+                TestCode = testCode
+            };
+
+            // test.ExpectedDiagnostics is an empty collection because ConfigureFunctionsWorkerDefaults() is not present
```

Comment:
Can you asset the collection is empty?

Reply:
The analyzer test infra internally does that. Meaning for this case, it is expecting Empty diagnostic collection. The test will fail if diagnostics are generated here (because it is expecting empty collection).